### PR TITLE
refactor: update cached identity if lost

### LINKED ISSUE #132: If the server does not recognize the identity, get another one
Related to the thread and answer here: https://arcxmoney.slack.com/archives/C04S7E03146/p1678658486130109?thread_ts=1678418938.980109&cid=C04S7E03146

This is a current possible flow:
1. User has an invalid `identityId` cached
2. Sdk initiates a Websocket connection
3. The server cannot find the `identityId` in the DB, so it drops the connection

The new flow should be:
1 to 3, same as above
4. The SDK clears the `identityId` in its cache
5. Get a new identity from the server
6. Try a websocket connection again. If the server fails again, log an error with `/log-sdk`. Otherwise, send requests as per usual.

### PR DESCRIPTION
**Pull request by xitronix**
Closes #132

### CODE REVIEW

**Review by gtupak — CHANGES_REQUESTED** [review]
The identity is indeed removed but requires a refresh. The second websockets connection still uses the old identity. See the recording below:

https://user-images.githubusercontent.com/5834876/225027857-2cb44f18-68f6-4393-b821-6737104c4574.mov

File: src/ArcxAnalyticsSdk.ts [hunk 1]
```diff
@@ -59,6 +59,19 @@ export class ArcxAnalyticsSdk {
 
     this._registerSocketListeners(socket)
 
+    socket.once('error', (error) => {
+      if (error.name === 'InternalServerError') {
```
Comment (gtupak) [inline src/ArcxAnalyticsSdk.ts:63]: Also, realized that in prod it's not an `InternalServerError` but rather a `BadRequestError`. See here: 

image

Identity ID: `32198a83d76a6993e501f643`

**Review by gtupak — CHANGES_REQUESTED** [review]

**Comment by xitronix** [comment]
https://user-images.githubusercontent.com/18648454/225289024-abcba570-3708-4645-b55f-30f8f01f8a39.mov

**Review by gtupak — APPROVED** [review]

**Pull request merged**